.github/workflows/frontend.yml

```
@@ -22,10 +22,14 @@ jobs:
             - name: Checkout repository
               uses: actions/checkout@v2
 
-            - name: Setup Node.js environment
+            - name: Read .nvmrc
+              run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+              id: nvm
+
+            - name: Setup Node.js ${{ steps.nvm.outputs.NODE_VERSION }}
               uses: actions/setup-node@v3
               with:
-                  node-version: '16.9.1'
+                  node-version: ${{ steps.nvm.outputs.NODE_VERSION }}
```

You do not need a separate step to read the file `.nvmrc`:
https://github.com/actions/setup-node/blob/v3.1.1/docs/advanced-usage.md#node-version-file

```suggestion
                  node-version-file: '.nvmrc'
```

---

Thanks! I didn't know about that option :+1: